cfme/tests/infrastructure/test_instance_analysis.py

```diff
@@ -166,68 +170,65 @@ def _host_remove_creds():
 
 @pytest.fixture(scope="module")
 def vm_name(provider, analysis_type):
-    vm_name = 'test_ssa_{}-{}'.format(fauxfactory.gen_alphanumeric(), analysis_type)
+    vm_name = 'test-ssa-{}-{}'.format(fauxfactory.gen_alphanumeric(), analysis_type)
     return vm_name
 
 
 @pytest.fixture(scope="module")
 def vm_analysis_data(provider, analysis_type):
-    base_data = provider.data.get('vm_analysis_new', {}).get('provisioning', {})
-    base_data.update(provider.data.get('vm_analysis_new', {}).get('vms', {}).get(analysis_type, {}))
-    return base_data
+
+    pdata = provider.data
+    provisioning_data = pdata.get('vm_analysis_new', {}).get('provisioning', {})
+
+    # Setup up the provisioning data for the instance/vm
+    # Will default to provisioning items under vm_analysis but if not defined
+    #   falls back to items under provider['provisioning'] key
+    if 'host' not in provisioning_data and 'host' in pdata['provisioning']:
```

Comment:
Consider using dict.update() or dict.setdefault() for most of these. Or iterate against has_key. 

update() just overwrites duplicate entries, and doesn't _quite_ match the behavior here. I think setdefault() is what you want, since it just returns if key is in the dictionary, but you have to iterate. Python 3.5 has a built-in for exactly this!

```python
provisioning_data.update(pdata['provisioning'])